src/components/DataGrid/DataGrid.tsx

```diff
@@ -0,0 +1,48 @@
+import { DataGrid as DataGridComponent, GridRowParams } from '@mui/x-data-grid';
+import { Box } from '@mui/material';
+import {
+  createGridColDef,
+  dataGridLocaleText,
+  inititialColumnVisibilityModel,
+  rowsWithId,
+} from './utils';
+import { useNavigate } from 'react-router-dom';
+import { DataGridProps } from '../../types/types';
+import DataGridToolbar from './Toolbar';
+
+export default function DataGrid({ data }: DataGridProps) {
+  const navigate = useNavigate();
+  const handleNavigate = (params: GridRowParams<any>) => () => {
+    navigate('/' + params.row.id);
```

Comment:
The cards have also a `onClick` that deals with navigating. Are both these methods needed?

Reply:
Yes, the handleNavigate of the `DataGrid` uses the route `/params.row.id` and the `Card` uses `/extract.uuid(zaaktype.url))`. The route of the datagrid also could be `/extract.uuid(params.row.url)`. But the `params` parameter is used to get the data of the row, since the `DataGrid` is not a `Array.map` of the zaaktypen but a MUI component

Reply:
Ohh I understand now. One navigate is used when the Zaaktypes are in the cards layout and the other when they are in the table layout.